cardano/src/block/block.rs

```diff
@@ -361,11 +361,11 @@ impl chain_core::property::Deserialize for Block {
 }
 
 impl chain_core::property::HasTransaction for Block {
-    type Transactions = Vec<TxAux>;
-    fn transactions(&self) -> Option<&Self::Transactions> {
+    type Transactions = [TxAux];
+    fn transactions(&self) -> &Self::Transactions {
         match self {
-            Block::BoundaryBlock(_) => None,
-            Block::MainBlock(ref blk) => Some(&blk.body.tx),
+            Block::BoundaryBlock(blk) => &blk.body.no_transactions,
```

Comment:
can't we return a static ref here `& []` instead of adding a un-necessary and confusing field to the boundary block ?

Reply:
Ah, of course!